PennyMe/ViewController.swift

```diff
@@ -108,6 +108,30 @@ class ViewController: UIViewController, UITextFieldDelegate {
         super.viewWillAppear(animated)
         // each time the view appears, check colours of the pins
         check_json_dict()
+        // check whether some setting has changed, if yes, reload all data on the map
+        if SettingsViewController.hasChanged {
+            addAnnotationsIteratively()
+            SettingsViewController.hasChanged = false
+        }
+        
+    }
+    
+    func addAnnotationsIteratively() {
+        let relevantUserDefauls : [String] = ["unvisitedSwitch", "visitedSwitch", "markedSwitch", "retiredSwitch"]
+        var includedStates : [String] = []
+        for userdefault in relevantUserDefauls {
+            if UserDefaults.standard.bool(forKey: userdefault) {
+                let partStr = String( userdefault.prefix(userdefault.count - 6))
+                includedStates.append(partStr)
+            }
+        }
+        PennyMap.removeAnnotations(artworks)
+        for artwork in artworks {
+            if includedStates.contains(artwork.status) {
+                PennyMap.addAnnotation(artwork)
+            }
```

Comment:
At the moment, it's still implemented such that all pins are removed and then reloaded  in a for loop. This could be improved by adding / removing only the ones that were changed, but then we would still need to iterate over all artworks to check their status, or we would need four separate lists / dictionaries, i.e. one "artworks" list per status. So I don't think it would be a big runtime improvement over the current version, but let me know if you have other ideas